test/Microsoft.AspNetCore.Routing.Tests/Template/TemplateMatcherTests.cs

```diff
@@ -991,6 +991,107 @@ public void TryMatch_MultipleOptionalParameters()
             Assert.False(values.ContainsKey("id"));
         }
 
+        [Theory]
+        [InlineData("///")]
+        [InlineData("/a//")]
+        [InlineData("/a/b//")]
+        [InlineData("//b//")]
+        [InlineData("///c")]
+        [InlineData("///c/")]
+        public void TryMatch_MultipleOptionalParameters_WithEmptyIntermediateSegmentsDoesNotMatch(string url)
```

Comment:
I'd like to see a few tests at a higher level than `TemplateMatcher` - ie tests that cover `Route` and attribute routing